flang/include/flang/Lower/FIRBuilder.h

```diff
@@ -69,8 +70,20 @@ class FirOpBuilder : public mlir::OpBuilder {
   mlir::Value convertWithSemantics(mlir::Location loc, mlir::Type toTy,
                                    mlir::Value val);
 
-  /// Get the entry block of the current Function
-  mlir::Block *getEntryBlock() { return &getFunction().front(); }
+  /// Get the block for adding Allocas. If OpenMP is enabled then get the
+  /// the alloca block from an Operation which can be Outlined. Otherwise
+  /// use the entry block of the current Function
+  mlir::Block *getAllocaBlock() {
```

Comment:
This is growing some header file dependencies and complexity. Can we move it to the `FIRBuilder.cpp` file?

Reply:
Done.